src/components/plots/barPlot.ts

```diff
@@ -151,6 +166,81 @@ export module Plot {
     }
 
     /**
+     * Retrieves the closest PlotData to queryPoint.
+     *
+     * Bars containing the queryPoint are considered closest. If queryPoint lies outside
+     * of all bars, we return the closest in the dominant axis (x for horizontal
+     * charts, y for vertical) and break ties using the secondary axis.
+     *
+     * @param {Point} queryPoint The point to which plot data should be compared
+     *
+     * @returns {PlotData} The PlotData closest to queryPoint
+     */
+    public getClosestPlotData(queryPoint: Point): PlotData {
+      var chartXExtent = { min: 0, max: this.width() };
+      var chartYExtent = { min: 0, max: this.height() };
+
+      var minPrimaryDist = Infinity;
+      var minSecondaryDist = Infinity;
+
+      var closestData: any[] = [];
+      var closestPixelPoints: Point[] = [];
+      var closestElements: Element[] = [];
+
+      var queryPtPrimary = this._isVertical ? queryPoint.x : queryPoint.y;
+      var queryPtSecondary = this._isVertical ? queryPoint.y : queryPoint.x;
+
+      this.datasetOrder().forEach((key) => {
+        var plotData = this.getAllPlotData(key);
+        plotData.pixelPoints.forEach((plotPt, i) => {
+          var bar = plotData.selection[0][i];
+
+          if (!Bar._intersectsBBox(chartXExtent, chartYExtent, bar.getBBox())) {
+            // bar isn't visible on plot; ignore it
+            return;
+          }
+
+          var primaryDist: number, secondaryDist: number;
+          if (Bar._intersectsBBox(queryPoint.x, queryPoint.y, bar.getBBox())) {
+            // queryPoint is inside of this bar; this is as close as it can be to the bar
+            primaryDist = -Infinity;
+            secondaryDist = -Infinity;
```

Comment:
This is a remnant of my previous hack at the problem. My intent was to differentiate between being *inside* of a bar and being on top of (zero-distance-from) a pixel point. Thinking on it again, however, it looks like the "on top of" case will trigger _intersectsBBox() anyway. Zero it is.